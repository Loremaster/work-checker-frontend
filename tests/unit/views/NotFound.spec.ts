import { mount, createLocalVue } from "@vue/test-utils";
import Vue from "vue";
import Vuetify from "vuetify";
import VueRouter from "vue-router";
import Component from "@/views/NotFound.vue";

const localVue = createLocalVue();
const router = new VueRouter();

localVue.use(VueRouter);

Vue.use(Vuetify);

describe("Authentication.vue", () => {
  const setup = () => {
    return mount(Component, {
      localVue,
      router,
    });
  };

  it("renders correctly", () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it("contains text about 404", () => {
    const wrapper = setup();
    expect(wrapper.text()).toContain("We don't know about this page yet");
  });

  it("redirect to root path after clicking button", () => {
    const wrapper = setup();
    const spy = jest.spyOn(router, "push");

    wrapper.find(".v-btn").trigger("click");

    expect(spy).toHaveBeenCalledWith(
      expect.objectContaining({
        path: "/",
      }),
    );
    spy.mockClear();
  });
});
