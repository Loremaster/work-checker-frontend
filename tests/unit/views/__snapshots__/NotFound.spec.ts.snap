// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication.vue renders correctly 1`] = `
<div class="container grid-list-md">
  <div class="layout row wrap">
    <div class="flex xs12">
      <div class="v-card v-sheet theme--light">
        <div class="v-card__title red darken-3 white--text subheading v-card__title--primary"><span class="title">Oops!</span></div>
        <div class="v-card__text">
          Looks like you're in uncharted territory. We don't know about this page yet. Luckily, we know the way back.
        </div>
        <div class="v-card__actions"><a href="#/" class="info v-btn--active v-btn v-btn--flat v-btn--router theme--light">
            <div class="v-btn__content">Back</div>
          </a></div>
      </div>
    </div>
  </div>
</div>
`;
